eZ/Bundle/EzPublishCoreBundle/Resources/config/services.yml

```diff
@@ -38,20 +38,55 @@ services:
         class: "%ezpublish.siteaccess.class%"
         arguments: ["%ezpublish.siteaccess.default.name%", 'uninitialized']
 
-    ezpublish.config.resolver.core:
-        public: true # @todo should be private
-        class: "%ezpublish.config.resolver.dynamic.class%"
-        arguments: ["@?logger", "%ezpublish.siteaccess.groups_by_siteaccess%", "%ezpublish.config.default_scope%"]
+    ezpublish.config.resolver.default:
+        public: true
+        class: eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver\DefaultScopeConfigResolver
+        arguments:
+            - '%ezpublish.config.default_scope%'
         calls:
-            - [setSiteAccess, ["@ezpublish.siteaccess"]]
-            - [setContainer, ["@service_container"]]
+            - [setContainer, ['@service_container']]
+        lazy: true
+        tags:
+            - { name: ezpublish.config.resolver, priority: 0 }
+
+    ezpublish.config.resolver.siteaccess_group:
+        class: eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver\SiteAccessGroupConfigResolver
+        arguments:
+            - '@ezpublish.siteaccess.provider'
+            - '%ezpublish.config.default_scope%'
+        calls:
+            - [setSiteAccess, ['@ezpublish.siteaccess']]
+            - [setContainer, ['@service_container']]
+        lazy: true
+        tags:
+            - { name: ezpublish.config.resolver, priority: 50 }
+
+    ezpublish.config.resolver.siteaccess_static:
+        class: eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver\StaticSiteAccessConfigResolver
+        arguments:
+            - '@ezpublish.siteaccess.provider'
+            - '%ezpublish.config.default_scope%'
+        calls:
+            - [setSiteAccess, ['@ezpublish.siteaccess']]
+            - [setContainer, ['@service_container']]
+        lazy: true
+        tags:
+            - { name: ezpublish.config.resolver, priority: 100 }
+
+    ezpublish.config.resolver.global:
+        public: true
+        class: eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver\GlobalScopeConfigResolver
+        arguments:
+            - '%ezpublish.config.default_scope%'
+        calls:
+            - [setContainer, ['@service_container']]
         lazy: true
         tags:
-            - { name: ezpublish.config.resolver, priority: 200 }
+            - { name: ezpublish.config.resolver, priority: 255 }
 
     ezpublish.config.resolver.chain:
         public: true # @todo should be private
-        class: "%ezpublish.config.resolver.chain.class%"
+        class: '%ezpublish.config.resolver.chain.class%'
```

Comment:
while at it, let's drop `.class` parameter, use class name in this line instead, and update bc doc (sorry, kinda falls into the scope of ChainConfigResolver improvements ;)).